src/main/java/gregtech/api/metatileentity/multiblock/MultiblockControllerBase.java

```diff
@@ -175,11 +186,18 @@ protected void checkStructurePattern() {
             ArrayList<IMultiblockPart> parts = new ArrayList<>(rawPartsSet);
             parts.sort(Comparator.comparing(it -> ((MetaTileEntity) it).getPos().hashCode()));
             for (IMultiblockPart part : parts) {
-                if (part.isAttachedToMultiBlock()) {
-                    //disallow sharing of multiblock parts
-                    //if part is already attached to another multiblock,
-                    //stop here without attempting to register abilities
-                    return;
+                if (!canShare()) {
+                    if (part.isAttachedToMultiBlock()) {
+
+                        return;
+                    }
+                }
+                else {
+                    if (part instanceof MetaTileEntityEnergyHatch || part instanceof MetaTileEntityRotorHolder) {
+                        if (part.isAttachedToMultiBlock()) {
+                            return;
+                        }
+                    }
```

Comment:
I think this could be cleaned up a bit
```suggestion
                if (part.isAttachedToMultiBlock()) {
                    if (!canShare() || part instanceof MetaTileEntityEnergyHatch || part instanceof MetaTileEntityRotorHolder) {
                        return;
                    }
                }
```